third_party/Outcome/include/outcome.hpp

```diff
@@ -0,0 +1,61 @@
+// Copyright (c) 2022 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef THIRD_PARTY_OUTCOME_OUTCOME_HPP_
+#define THIRD_PARTY_OUTCOME_OUTCOME_HPP_
+
+// This header is a shim layer which takes the Boost version of Outcome and makes it behave like the
+// standalone version. It implements just enough to make Orbit compile with Boost..
+//
+// It also tries to keep Boost headers and libraries out of the include and dependency graph.
+// Only outcome headers from Boost are required. No linking against Boost libraries.
+
+// This will not be used because we always use the terminate policy, but it is needed to
+// consume the following includes.
+#define BOOST_OUTCOME_THROW_EXCEPTION(...) std::abort();
+
+#include <boost/outcome/config.hpp>
+#include <boost/outcome/std_result.hpp>
+#include <boost/outcome/success_failure.hpp>
+#include <boost/outcome/try.hpp>
+#include <system_error>
+#include <utility>
+
+#define OUTCOME_V2_NAMESPACE outcome_v2
```

Comment:
Why do we need this?

Reply:
Outcome generates a version-dependent namespace and only exposes this macro. So we need to do the same.

You can check out OrbitBase/Result.h for how it works in detail.